common/project-manager/src/main/scala/org/enso/projectmanager/boot/ZioPlatform.scala

```diff
@@ -0,0 +1,41 @@
+package org.enso.projectmanager.boot
+
+import zio.Cause
+import zio.internal.stacktracer.Tracer
+import zio.internal.stacktracer.impl.AkkaLineNumbersTracer
+import zio.internal.tracing.TracingConfig
+import zio.internal.{Executor, Platform, Tracing}
+
+import scala.concurrent.ExecutionContext
+
+/**
+  * An environment needed to execute ZIO actions.
+  *
+  * @param computeExecutionContext compute thread pool
+  */
+class ZioPlatform(computeExecutionContext: ExecutionContext) extends Platform {
+
+  override def executor: Executor =
+    Executor.fromExecutionContext(2048)(computeExecutionContext)
+
+  override val tracing = Tracing(
+    Tracer.globallyCached(new AkkaLineNumbersTracer),
+    TracingConfig.enabled
+  )
+
+  override def fatal(t: Throwable): Boolean =
+    t.isInstanceOf[VirtualMachineError]
+
+  override def reportFatal(t: Throwable): Nothing = {
+    t.printStackTrace()
+    try {
+      System.exit(-1)
+      throw t
+    } catch { case _: Throwable => throw t }
```

Comment:
you could use `finally { throw t }` instead of the double-throw, seems less confusing.
```suggestion
    try {
      System.exit(-1)
    } finally { throw t }
```

Reply:
I applied this change, but it doesn't compile. I copied this code from ZIO lib. It is a hack to return Nothing.

Reply:
Could you not just return Nothing at the end?

Reply:
No, it is ZIO API.

Reply:
In that case: ¯\_(ツ)_/¯